ludwig/backend/ray.py

```diff
@@ -501,6 +512,12 @@ def train(
         # use `strict=False` to account for PEFT training, where the saved state in the checkpoint
         # might only contain the PEFT layers that were modified during training
         state_dict, *args = results
+
+        # Re-add the model weights to the model
+        model, model_weights = ray.get(model_ref)
+        replace_tensors(model, model_weights, torch.device("cpu"))
+        self.model = model
+        self.model.initialize_adapter()
```

Comment:
this needs to be a gated step (i.e. check for if the model is an LLM or otherwise)